kolibri/core/assets/src/vue/content-renderer/download-button.vue

```diff
@@ -58,20 +58,16 @@
       downloadMediaText() {
         return this.$tr('downloadMedia');
       },
+      downloadLink() {
+        const filePath = this.availableFiles[0].storage_url;
+        // Extract filename from the storage url.
+        const filename = filePath.replace(/^.*[\\\/]/, '');
```

Comment:
The filename can be constructed from `checksum + "." + extension`, if that seems cleaner. Otherwise, another approach that is (possibly?) cleaner would be `.split("/").slice(-1)[0]`